src/lib.rs

```diff
@@ -161,21 +169,21 @@ impl Element {
             namespaces: None,
             attributes: HashMap::new(),
             children: Vec::new(),
-            text: None,
         }
     }
 
     /// Parses some data into an Element
-    pub fn parse<R: Read>(r: R) -> Result<Element, ParseError> {
+    pub fn parse<R: Read>(r: R) -> Result<Vec<XMLNode>, ParseError> {
```

Comment:
If at all possible, I'd like to keep the original signature and return only a single `Element`, and not a `Vec<XMLNode>`.